### PR TITLE
Utilize retrofit rxjava(1) adapter

### PR DESCRIPTION
**Pull request by DavidJFelix**
- Rather than binding rxjava to retrofit call structure, it
  is more idiomatic to use the rxjava adapter for retrofit
  and handle errors through its normal onError for the observable

- **TESTS DO NOT PASS**. This is a POC commit for a blog tutorial repo.
  I did not take the time to change the tests and make them pass.
  The mocked out calls will need to change and the way errors are
  detected will need to follow the normal observable pattern.

### CODE REVIEW

File: app/src/main/java/com/droidcba/kedditbysteps/features/news/NewsManager.kt [hunk 1]
```diff
@@ -23,28 +24,21 @@ class NewsManager @Inject constructor(private val api: NewsAPI) {
...
-                        dataResponse.after ?: "",
-                        dataResponse.before ?: "",
-                        news)
 
-                subscriber.onNext(redditNews)
-                subscriber.onCompleted()
-            } else {
-                subscriber.onError(Throwable(response.message()))
-            }
+    fun extractNewsFromResponse(b: RedditNewsResponse): RedditNews {
```
Comment (DavidJFelix) [inline app/src/main/java/com/droidcba/kedditbysteps/features/news/NewsManager.kt:32]: I forgot to change the variable name here.

**Review by juanchosaravia — CHANGES_REQUESTED** [review]
Thanks @DavidJFelix for the PR!
In order to merge this PR I need the tests to be fixed. If you find the time to do it I'll really appreciate it, otherwise I'll leave it in this way after I find the time to update this and also the tutorial.

**Comment by DavidJFelix** [comment]
I may, as I get around to testing my own app. I really appreciated the blog/tutorial in getting kotlin working in my own app, so when I saw mixing rxjava and retrofit manually I just had to give back. I'll keep you updated if I figure out how to test the rxjava'd code the way you were testing the retrofit calls.

**Comment by danielgomezrico** [comment]
@DavidJFelix check out this class:

```
class BaseTest {

  // Call on Each @Before
  open fun setUp() {
    mockSchedulers()
  }

  fun mockSchedulers() {
    resetSchedulersMock()

    RxAndroidPlugins.getInstance().registerSchedulersHook(object : RxAndroidSchedulersHook() {
      override fun getMainThreadScheduler() = Schedulers.immediate()
    })

    RxJavaHooks.setOnIOScheduler { Schedulers.immediate() }
    RxJavaHooks.setOnComputationScheduler { Schedulers.immediate() }
    RxJavaHooks.setOnNewThreadScheduler { Schedulers.immediate() }
  }

  fun resetSchedulersMock() {
    RxJavaHooks.reset()
    RxAndroidPlugins.getInstance().reset()
  }
}
```

You need to call `setUp` on `@Before` for each unit test in order to use the immediate scheduler and avoid going to other threads in functions using observables.

Hope it helps in your tests

**Comment by juanchosaravia** [comment]
Using Coroutines so closing this PR

**Pull request closed without merging**